File: kgstatsSrv/src/main/java/ru/klavogonki/statistics/export/StatisticsGeneratorConfig.java
package ru.klavogonki.statistics.export;

import lombok.Data;

/**
 * <p>
 * Конфиг, настраивающий генерацию словарей.
 * </p>
 *
 * <p>
 * todo: по факту, в конфиг можно перенести настройки, которые сейчас реализуют классы,
 * реализующие VocabularyTopExporter.
 * </p>
 */
@Data
public class StatisticsGeneratorConfig {

    // aggregate tops
    private boolean exportIndexPage;

    private boolean exportTopBySpeed; // old exporter, not by normal vocabulary

    private boolean exportTop500Pages;

    private boolean exportPlayersByRank;

    // standard vocabularies tops
    private boolean exportNormalTop;
    private boolean exportAbraTop;
    private boolean exportReferatsTop;
    private boolean exportNoErrorTop;
    private boolean exportMarathonTop;
    private boolean exportCharsTop;
    private boolean exportDigitsTop;
    private boolean exportSprintTop;

    // non-standard vocabularies tops
    private boolean exportNormalInEnglishTop;
    private boolean exportMiniMarathonTop;
    private boolean exportShortTextsTop;
    private boolean exportFrequencyVocabularyTop;
    private boolean exportOneHundredRussianTop;
    private boolean exportDigitsOneHundredTop;
    private boolean exportTrainingIndexFingersTop;
    private boolean exportRingFingersTop;
    private boolean exportPinkiesPlusTop;

    // multi-vocabularies tops
    private boolean exportMultiLinguaTop;
}

File: kgstatsSrv/src/main/kotlin/ru/klavogonki/statistics/export/StatisticsGeneratorConfig.kt
package ru.klavogonki.statistics.export

/**
 * Конфиг, настраивающий генерацию словарей.
 *
 * todo: по факту, в конфиг можно перенести настройки, которые сейчас реализуют классы,
 * реализующие [ru.klavogonki.statistics.export.vocabulary.VocabularyTopExporter].
 */
data class StatisticsGeneratorConfig(
    // aggregate tops
    val exportIndexPage: Boolean = false,

    val exportTopBySpeed: Boolean = false, // old exporter, not by normal vocabulary
    val exportTop500Pages: Boolean = false,
    val exportPlayersByRank: Boolean = false,

    // standard vocabularies
    val standardDictionariesCodes: List<String> = emptyList(),

    // non-standard vocabularies
    val nonStandardDictionariesCodes: List<Int> = emptyList()
)